### PR TITLE
Add COWC counting experiment

### PR DESCRIPTION
**Pull request by adamjstewart**
Original authors got RMSE: 0.657, MAE: 0.248 with ResCeption, I got RMSE: 0.597, MAE: 0.290 with ResNet-50

Optimal hyperparams are pre-trained ResNet-50 with a learning rate of 0.0001. Still need to run seed experiments though.

### CODE REVIEW

**Comment by adamjstewart** [comment]
Results for all hyperparam runs:
```console
$ python experiments/find_optimal_hyperparams.py 
{
    "resnet18_0.0001_False": {
        "test": {
            "MAE": 0.4228631854057312,
            "RMSE": 0.8189011216163635
        },
        "train": {
            "MAE": 0.06813093274831772,
            "RMSE": 0.10447093844413757
        },
        "val": {
            "MAE": 0.30923041701316833,
            "RMSE": 0.5510013699531555
        }
    },
    "resnet18_0.0001_True": {
        "test": {
            "MAE": 0.3503802716732025,
            "RMSE": 0.6810266971588135
        },
        "train": {
            "MAE": 0.045050643384456635,
            "RMSE": 0.0662403553724289
        },
        "val": {
            "MAE": 0.23932622373104095,
            "RMSE": 0.4339234530925751
        }
    },
    "resnet18_0.001_False": {
        "test": {
            "MAE": 0.34669458866119385,
            "RMSE": 0.7023187279701233
        },
        "train": {
            "MAE": 0.0947667583823204,
            "RMSE": 0.1607394516468048
        },
        "val": {
            "MAE": 0.2499205619096756,
            "RMSE": 0.46694502234458923
        }
    },
    "resnet18_0.001_True": {
        "test": {
            "MAE": 0.31933149695396423,
            "RMSE": 0.6337572932243347
        },
        "train": {
            "MAE": 0.10042704641819,
            "RMSE": 0.1720120757818222
        },
        "val": {
            "MAE": 0.24263297021389008,
            "RMSE": 0.45746302604675293
        }
    },
    "resnet18_0.01_False": {
        "test": {
            "MAE": 0.38065099716186523,
            "RMSE": 0.7340189218521118
        },
        "train": {
            "MAE": 0.12802830338478088,
            "RMSE": 0.211224764585495
        },
        "val": {
            "MAE": 0.3020056188106537,
            "RMSE": 0.5527365207672119
        }
    },
    "resnet18_0.01_True": {
        "test": {
            "MAE": 0.393193781375885,
            "RMSE": 0.7751938700675964
        },
        "train": {
            "MAE": 0.1762107014656067,
            "RMSE": 0.29141849279403687
        },
        "val": {
            "MAE": 0.3135584592819214,
            "RMSE": 0.5685145854949951
        }
    },
    "resnet50_0.0001_False": {
        "test": {
            "MAE": 0.3953153192996979,
            "RMSE": 0.7945008277893066
        },
        "train": {
            "MAE": 0.1251937299966812,
            "RMSE": 0.20272821187973022
        },
        "val": {
            "MAE": 0.28527218103408813,
            "RMSE": 0.5174267888069153
        }
    },
    "resnet50_0.0001_True": {
        "test": {
            "MAE": 0.289704293012619,
            "RMSE": 0.5968681573867798
        },
        "train": {
            "MAE": 0.03914037346839905,
            "RMSE": 0.061728719621896744
        },
        "val": {
            "MAE": 0.1886562556028366,
            "RMSE": 0.3793787956237793
        }
    },
    "resnet50_0.001_False": {
        "test": {
            "MAE": 0.30080103874206543,
            "RMSE": 0.6167911887168884
        },
        "train": {
            "MAE": 0.11078942567110062,
            "RMSE": 0.20426179468631744
        },
        "val": {
            "MAE": 0.21907347440719604,
            "RMSE": 0.4304286241531372
        }
    },
    "resnet50_0.001_True": {
        "test": {
            "MAE": 0.28484395146369934,
            "RMSE": 0.5833998918533325
        },
        "train": {
            "MAE": 0.0910743772983551,
            "RMSE": 0.17132015526294708
        },
        "val": {
            "MAE": 0.21179042756557465,
            "RMSE": 0.4243508279323578
        }
    },
    "resnet50_0.01_False": {
        "test": {
            "MAE": 0.35612013936042786,
            "RMSE": 0.6897898316383362
        },
        "train": {
            "MAE": 0.2029591053724289,
            "RMSE": 0.3481382131576538
        },
        "val": {
            "MAE": 0.29972490668296814,
            "RMSE": 0.5552846193313599
        }
    },
    "resnet50_0.01_True": {
        "test": {
            "MAE": 0.3393082320690155,
            "RMSE": 0.6614287495613098
        },
        "train": {
            "MAE": 0.17775976657867432,
            "RMSE": 0.3098694086074829
        },
        "val": {
            "MAE": 0.2752934992313385,
            "RMSE": 0.5165057182312012
        }
    }
}
```
So pre-trained lr=0.0001 is optimal for both ResNet-18 and ResNet-50. Will start running seed experiments.

**Comment by adamjstewart** [comment]
Results of seed experiments:
```console
$ python experiments/find_optimal_hyperparams.py 
{
    "resnet18_0.0001_True_0": {
        "test": {
            "MAE": 0.3500770628452301,
            "RMSE": 0.6806954145431519
        },
        "train": {
            "MAE": 0.043942272663116455,
            "RMSE": 0.06442411988973618
        },
        "val": {
            "MAE": 0.2380620241165161,
            "RMSE": 0.4320961833000183
        }
    },
    "resnet18_0.0001_True_1": {
        "test": {
            "MAE": 0.3450622260570526,
            "RMSE": 0.6763545274734497
        },
        "train": {
            "MAE": 0.03702713921666145,
            "RMSE": 0.056145668029785156
        },
        "val": {
            "MAE": 0.23265080153942108,
            "RMSE": 0.4271908104419708
        }
    },
    "resnet18_0.0001_True_2": {
        "test": {
            "MAE": 0.34744811058044434,
            "RMSE": 0.6652932167053223
        },
        "train": {
            "MAE": 0.04200323298573494,
            "RMSE": 0.06205281242728233
        },
        "val": {
            "MAE": 0.23928424715995789,
            "RMSE": 0.42992401123046875
        }
    },
    "resnet18_0.0001_True_3": {
        "test": {
            "MAE": 0.3235655426979065,
            "RMSE": 0.6568469405174255
        },
        "train": {
            "MAE": 0.02764858491718769,
            "RMSE": 0.041858747601509094
        },
        "val": {
            "MAE": 0.2157038450241089,
            "RMSE": 0.41813915967941284
        }
    },
    "resnet18_0.0001_True_4": {
        "test": {
            "MAE": 0.33927392959594727,
            "RMSE": 0.6693754196166992
        },
        "train": {
            "MAE": 0.0394032746553421,
            "RMSE": 0.05985004082322121
        },
        "val": {
            "MAE": 0.22781652212142944,
            "RMSE": 0.4261322617530823
        }
    },
    "resnet18_0.0001_True_5": {
        "test": {
            "MAE": 0.3433762788772583,
            "RMSE": 0.6663753390312195
        },
        "train": {
            "MAE": 0.04274104908108711,
            "RMSE": 0.06269002705812454
        },
        "val": {
            "MAE": 0.23423224687576294,
            "RMSE": 0.42900359630584717
        }
    },
    "resnet18_0.0001_True_6": {
        "test": {
            "MAE": 0.3449496924877167,
            "RMSE": 0.6690500378608704
        },
        "train": {
            "MAE": 0.04379786178469658,
            "RMSE": 0.06545503437519073
        },
        "val": {
            "MAE": 0.23480984568595886,
            "RMSE": 0.4279843866825104
        }
    },
    "resnet18_0.0001_True_7": {
        "test": {
            "MAE": 0.34038010239601135,
            "RMSE": 0.6671060919761658
        },
        "train": {
            "MAE": 0.037459805607795715,
            "RMSE": 0.05588199943304062
        },
        "val": {
            "MAE": 0.22864103317260742,
            "RMSE": 0.4235287010669708
        }
    },
    "resnet18_0.0001_True_8": {
        "test": {
            "MAE": 0.33597931265830994,
            "RMSE": 0.6571869254112244
        },
        "train": {
            "MAE": 0.038087114691734314,
            "RMSE": 0.057052940130233765
        },
        "val": {
            "MAE": 0.22788715362548828,
            "RMSE": 0.4241137206554413
        }
    },
    "resnet18_0.0001_True_9": {
        "test": {
            "MAE": 0.3303143382072449,
            "RMSE": 0.6588548421859741
        },
        "train": {
            "MAE": 0.038068510591983795,
            "RMSE": 0.05842827260494232
        },
        "val": {
            "MAE": 0.22235584259033203,
            "RMSE": 0.42193105816841125
        }
    },
    "resnet50_0.0001_True_0": {
        "test": {
            "MAE": 0.2704925537109375,
            "RMSE": 0.5821202993392944
        },
        "train": {
            "MAE": 0.021246742457151413,
            "RMSE": 0.03512001037597656
        },
        "val": {
            "MAE": 0.17061282694339752,
            "RMSE": 0.3671601116657257
        }
    },
    "resnet50_0.0001_True_1": {
        "test": {
            "MAE": 0.2648864984512329,
            "RMSE": 0.5677511096000671
        },
        "train": {
            "MAE": 0.0217603649944067,
            "RMSE": 0.03571641072630882
        },
        "val": {
            "MAE": 0.16968221962451935,
            "RMSE": 0.3672628402709961
        }
    },
    "resnet50_0.0001_True_2": {
        "test": {
            "MAE": 0.2668568193912506,
            "RMSE": 0.5714248418807983
        },
        "train": {
            "MAE": 0.02340245619416237,
            "RMSE": 0.03768444061279297
        },
        "val": {
            "MAE": 0.1711343675851822,
            "RMSE": 0.3664420247077942
        }
    },
    "resnet50_0.0001_True_3": {
        "test": {
            "MAE": 0.26749056577682495,
            "RMSE": 0.573369562625885
        },
        "train": {
            "MAE": 0.02169989049434662,
            "RMSE": 0.03456055000424385
        },
        "val": {
            "MAE": 0.17127622663974762,
            "RMSE": 0.368755042552948
        }
    },
    "resnet50_0.0001_True_4": {
        "test": {
            "MAE": 0.2659497857093811,
            "RMSE": 0.5699235200881958
        },
        "train": {
            "MAE": 0.0215201023966074,
            "RMSE": 0.03461863473057747
        },
        "val": {
            "MAE": 0.16829906404018402,
            "RMSE": 0.36388155817985535
        }
    },
    "resnet50_0.0001_True_5": {
        "test": {
            "MAE": 0.27048465609550476,
            "RMSE": 0.5762578845024109
        },
        "train": {
            "MAE": 0.02540701813995838,
            "RMSE": 0.04083617031574249
        },
        "val": {
            "MAE": 0.1724844127893448,
            "RMSE": 0.3702339231967926
        }
    },
    "resnet50_0.0001_True_6": {
        "test": {
            "MAE": 0.2673390805721283,
            "RMSE": 0.5666937828063965
        },
        "train": {
            "MAE": 0.02274135686457157,
            "RMSE": 0.036585282534360886
        },
        "val": {
            "MAE": 0.1721368432044983,
            "RMSE": 0.36737993359565735
        }
    },
    "resnet50_0.0001_True_7": {
        "test": {
            "MAE": 0.26657676696777344,
            "RMSE": 0.5658478736877441
        },
        "train": {
            "MAE": 0.023826995864510536,
            "RMSE": 0.03880121931433678
        },
        "val": {
            "MAE": 0.17266304790973663,
            "RMSE": 0.36689385771751404
        }
    },
    "resnet50_0.0001_True_8": {
        "test": {
            "MAE": 0.27597469091415405,
            "RMSE": 0.57489413022995
        },
        "train": {
            "MAE": 0.029649924486875534,
            "RMSE": 0.04738578572869301
        },
        "val": {
            "MAE": 0.1786545068025589,
            "RMSE": 0.36937782168388367
        }
    },
    "resnet50_0.0001_True_9": {
        "test": {
            "MAE": 0.26575517654418945,
            "RMSE": 0.5771275162696838
        },
        "train": {
            "MAE": 0.01985890604555607,
            "RMSE": 0.032478779554367065
        },
        "val": {
            "MAE": 0.16781100630760193,
            "RMSE": 0.3684160113334656
        }
    }
}
```

**Comment by adamjstewart** [comment]
So, final results to add to the table in our paper:

| Dataset  | Method | Weight Initialization | Bands | Performance |
| ------------- | ------------- | --- | --- | --- |
| COWC Counting | ResNet-50 | ImageNet | RGB | **0.573 ± 0.005** |
| COWC Counting | ResNet-18 | ImageNet | RGB | 0.667 ± 0.007 |
| COWC Counting | ResCeption | random? | RGB | 0.657 |

**Comment by adamjstewart** [comment]
This is with no data augmentation, we could also add random flips and things to improve performance.

**Review by calebrob6 — APPROVED** [review]
